crud/common/utils.lua

```diff
@@ -711,4 +713,55 @@ function utils.update_storage_call_error_description(err, func_name, replicaset_
     return err
 end
 
+--- Polls replicas for storage state
+--
+-- @function get_storages_state
+--
+-- @tparam ?number opts.timeout
+--  Function call timeout
+--
+-- @return a table of storage states by replica uuid.
+function utils.get_storages_state(opts)
+    local replicasets, err = vshard.router.routeall()
+    if replicasets == nil then
+        return nil, GetReplicaStateError:new("Failed to get all replicasets: %s", err.err)
+    end
+
+    opts = opts or {}
+
+    local futures_by_replicas = {}
+    local replicas_state = {}
+    local async_opts = {is_async = true}
+    local timeout = opts.timeout or const.DEFAULT_VSHARD_CALL_TIMEOUT
+
+    for _, replicasets in pairs(replicasets) do
+        for replica_uuid, replica in pairs(replicasets.replicas) do
+            replicas_state[replica_uuid] = {name = replica.name, storage_initialized = false}
+            local ok, future = pcall(replica.conn.eval, replica.conn,
+                                [[return rawget(rawget(_G, "_crud") or {}, "is_init") ~= nil]],
+                                {}, async_opts)
```

Comment:
Please, use `call`. You need a function on the storage side, yep.

`eval` assumes ability to do everything. We should not require it to call the status check function.

Reply:
So, we will must process "undefined function" error, which will be returned if a storage is not initialized yet. What is the point of calling the function if the fact of its existence means initialized storage?